### PR TITLE
Merge cp4waiops-sandbox into cp4waiops-gitops

### PR DESCRIPTION
**Pull request by morningspace**
This PR is aimed to merge configuration from [cp4waiops-sandbox](https://github.com/morningspace/cp4waiops-sandbox) into [cp4waiops-gitops](https://github.com/morningspace/cp4waiops-gitops). It includes:

* The all-in-one helm chart
* The argocd customization helm chart
* The fyre sample cluster helm chart
* The cp4waiops custom helm chart that supports x-small profile

It also:

* Moved ceph configuration into config folder
* Dropped the unused environments folder

TODO:

- [x] Revert repoURL and targetRevision changes
- [ ] Support custom build in all-in-one
- [ ] Support event manager in all-in-one
- [x] ir taskmanager & jobmanager keep restarting

### CODE REVIEW

File: config/all-in-one/templates/cluster-operator-fyre.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{{- if and .Values.cluster.enabled }}
```
Comment (gyliu513) [inline config/all-in-one/templates/cluster-operator-fyre.yaml:1]: Here may also need to guarantee that the provder.type is fyre.

File: config/all-in-one/values.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+# Cluster parameters
+# -----------------------------------------------------------------------------
+cluster:
+
+  # Specify whether or not to provision a cluster before install CP4WAIOps.
+  enabled: false
+
+  # Provider specific configuration
+  provider:
+    # Fyre configuration
```
Comment (gyliu513) [inline config/all-in-one/values.yaml:28]:
```suggestion
    # Fyre configuration
    # NOTE: Fyre is an internal used IaaS platform for IBM
```

File: config/all-in-one/values.yaml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    fyre:
```
Comment (gyliu513) [inline config/all-in-one/values.yaml:29]: We may need to think more about this: How to enable the checking for different providers with some condition checking.

File: config/all-in-one/values.yaml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  version: "3.2"
+
+  # The username of image registry used to pull CP4WAIOps images.
+  dockerUsername: cp
+
+  # The password of image registry used to pull CP4WAIOps images.
+  dockerPassword: REPLACE_IT
+
+  # The CP4WAIOps deployment profile, e.g.: x-small, small, large.
+  profile: x-small
```
Comment (gyliu513) [inline config/all-in-one/values.yaml:65]: official release do not have `x-small` yet, but only `small` and `large`, can we highlight `x-small` is only for small PoC?

File: config/all-in-one/values.yaml (hunk 3 above)
Comment (morningspace) [reply config/all-in-one/values.yaml:29]: Sure, this can be supported when we have more providers to come in.

**Comment by morningspace** [comment]
@gyliu513 @liyanwei93 This PR is ready for review now. For the remaining TODOs that I listed in issue summary, I will complete the task "Revert repoURL and targetRevision changes" once review is passed. For other tasks, I will raise separate PRs to address:
- Support custom build in all-in-one
- Support event manager in all-in-one

**Review by gyliu513 — APPROVED** [review]

**Review by liyanwei93 — APPROVED** [review]

**Pull request merged**